src/Compiler/Checking/MethodCalls.fs

```diff
@@ -1237,7 +1237,8 @@ let MethInfoChecks g amap isInstance tyargsOpt objArgs ad m (minfo: MethInfo)  =
                 AccessibleFrom(paths, None) 
         | _ -> ad
 
-    if not (IsTypeAndMethInfoAccessible amap m adOriginal ad minfo) then 
+    let isProtectedSetter = minfo.IsProtectedAccessibility && minfo.LogicalName.StartsWithOrdinal("set_")
+    if not (IsTypeAndMethInfoAccessible amap m adOriginal ad minfo) && not isProtectedSetter then 
```

Comment:
I wonder if it makes sense to inline the check, so it's only performed if LHS is `true`?

```fsharp
if false && (printfn "side-effect"; true) then printfn "yes";;
```

Reply:
Sure. Thanks for the review :)